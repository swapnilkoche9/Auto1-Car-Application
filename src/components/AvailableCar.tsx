import React from 'react'
import { Link } from 'react-router-dom'
import constants from '../constants/SystemConstants'
import { AvailableCarProps } from '../utils/interface'

const AvailableCar = (props: AvailableCarProps) => (
  <div className='availableCar'>
    <div className='carImageContainer'>
      <img className='carImage' src={props.pictureUrl} alt="car" />
    </div>
    <div className='availableCarDetails'>
      <div>
        <p className='availableCarstext availableCarsTextBold'>{`${props.modelName}`}</p>
      </div>
      <div className='availableCarstext availableCarsTextSmall'>
        <p>{`${constants.STOCK_LABEL}
            ${props.stockNumber} -
            ${props.mileage.number} ${props.mileage.unit.toUpperCase()} -
            ${props.fuelType} -
            ${props.color}`}</p>
      </div>
      <div className="viewDetailsLinkContainer">
        <Link to={`/car/${props.stockNumber}`} className='availableCarstext availableCarsTextSmall linkTextOange'>{constants.VIEW_DETAILS_TEXT}</Link>
      </div>
    </div>
  </div>
)

export default AvailableCar
