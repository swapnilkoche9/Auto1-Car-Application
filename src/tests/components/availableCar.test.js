import React from 'react'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import AvailableCar from '../../components/AvailableCar'

Enzyme.configure({ adapter: new Adapter() })
const { shallow } = Enzyme

const getPageParams = jest.fn()
const wrapper = shallow(
  <AvailableCar
    getPageParams={getPageParams}
    stockNumber={100}
    mileage={{ number: 100, unit: 'km' }}
    fuelTye={'Petrol'}
    color={'Yellow'}
    modelName={'Taurus'}
  />
)
test('should render AvailableCar correctly', () => {
  expect(wrapper).toMatchSnapshot()
})

test('should render AvailableCar with alt data correctly', () => {
  wrapper.setProps({
    color: 'block'
  })
  expect(wrapper).toMatchSnapshot()
})
