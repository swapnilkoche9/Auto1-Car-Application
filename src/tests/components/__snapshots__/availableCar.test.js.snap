// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AvailableCar correctly 1`] = `ShallowWrapper {}`;

exports[`should render AvailableCar with alt data correctly 1`] = `ShallowWrapper {}`;
